File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        mp = {}; # dict
        for i,x in enumerate(nums):
            f = target-nums[i]
            if mp.get(f) != None:
                ans = ans + [mp.get(f), i]
                return ans
            mp[x] = i

File: 1-two-sum/1-two-sum.java
import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int[] ans = new int[2];
        HashMap<Integer, Integer> mp = new HashMap<Integer, Integer>();
        for(int i=0; i<nums.length; ++i){
            int f= target-nums[i];
            if(mp.get(f)!=null){ 
                
                ans[0] = i;
                ans[1] = mp.get(f);
                
                return ans;
            }
            
            mp.put(nums[i], i);
        }
        return ans;
    }
}